### DISCUSSION TITLE
I made a snap for Tribler

### DISCUSSION THREAD

**Discussion by Hans2004** [Tribler/tribler#5976] [General]
I have made a snap for Tribler version 7.7.1 that runs great on, at least, OpenSuse Tumbleweed and Ubuntu 18.04

If you have snapcraft installed you can try it by typing:

snap install --edge test-tribler

A few seconds later, you should have tribler installed on your system and you can run it from your start menu.

In the past I have tried running tribler directly on OpenSuse, which was insanely difficult because it required recompiling python against an alternate compiled openssl that had elliptic curves enabled.

Then I made a docker image that connected with X11, and used that for a while. Problem there was that after a while the tribler gui would not show on my system anymore (don’t know why).

Then I made a docker image that I could connect to with VLC. This worked fine, but was quite the resource hog.

For some time the 32 bit tribler windows version ran fine on wine on my OpenSuse box. But the newer versions have issues.

This snap runs great and super stable on my OpenSuse tumbleweed, and on kubuntu 18.04 in a virtual machine. I hope it will run well on other systems as well. Perhaps this can motivate some people on the development team to release future versions of tribler on snapcraft as well.

I believe that this way of packaging tribler is very important, especially in the case of tribler because it would essentially double the exposure of tribler on linux systems.

**Comment by xoriole** [comment]
@Hans2004 thank you for starting the discussion here. It is good to see you trying Tribler in different linux distributions. Also, great to know that you managed to run snap package on linux. If you would like to contribute by creating a pull request to better/faster build snap package that would be great. 👍

**Reply by Hans2004** [reply]
I am not sure what you mean by creating a pull request (I don't know what a pull request is or does). Please explain what you want me to do. I am happy to share access to my github repo where I store the snapcraft.yaml file. That way your team can work with it. I was actually hoping that one of the students working on programming and packaging tribler will be interested in taking over the process of creating snaps for tribler. This might be of benefit for their future career. I don't work in IT professionally (I do create programs and scripts when I need something that is not available or too expensive commercially).
